plugins/modules/ntnx_service_groups.py

```diff
@@ -0,0 +1,238 @@
+from ..module_utils.utils import remove_param_with_none_value  # noqa: E402
+
+
+def get_module_spec():
+
+    icmp_spec = dict(code=dict(type="int"), type=dict(type="int"))
+
+    service_spec = dict(
+        tcp=dict(type="list", elements="str"),
+        udp=dict(type="list", elements="str"),
+        icmp=dict(
+            type="list",
+            elements="dict",
+            options=icmp_spec,
+        ),
+        any_icmp=dict(type="bool", default=False),
+    )
+
+    module_args = dict(
+        name=dict(type="str"),
+        desc=dict(type="str"),
+        service_group_uuid=dict(type="str"),
+        services=dict(
+            type="dict", options=service_spec, mutually_exclusive=[("icmp", "any_icmp")]
+        ),
+    )
+
+    return module_args
+
+
+def create_service_group(module, result):
+    service_group = ServiceGroup(module)
+    spec, error = service_group.get_spec()
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating service_groups spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        return
+
+    resp = service_group.create(spec)
+    service_group_uuid = resp["uuid"]
+    result["changed"] = True
+    result["response"] = resp
+    result["service_group_uuid"] = service_group_uuid
+
+
+def update_service_group(module, result):
+    service_group = ServiceGroup(module)
+    service_group_uuid = module.params.get("service_group_uuid")
+    if not service_group_uuid:
+        result["error"] = "Missing parameter service_group_uuid in playbook"
+        module.fail_json(msg="Failed updating service_group", **result)
+    result["service_group_uuid"] = service_group_uuid
+
+    # read the current state of service_group
+    resp = service_group.read(service_group_uuid)
+    resp = resp.get("service_group")
+
+    # new spec for updating service_group
+    update_spec, error = service_group.get_spec(resp)
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating service_group update spec", **result)
+
+    # check for idempotency
+    if resp == update_spec:
+        result["skipped"] = True
+        module.exit_json(
+            msg="Nothing to change. Refer docs to check for fields which can be updated"
+        )
+
+    if module.check_mode:
+        result["response"] = update_spec
+        return
+
+    # update service_group
+    resp = service_group.update(update_spec, uuid=service_group_uuid, no_response=True)
+
+    result["changed"] = True
+    result["response"] = resp
+
+
+def delete_service_group(module, result):
+    service_group_uuid = module.params["service_group_uuid"]
+    if not service_group_uuid:
+        result["error"] = "Missing parameter service_group_uuid in playbook"
+        module.fail_json(msg="Failed deleting service_groups", **result)
+
+    service_group = ServiceGroup(module)
+    resp = service_group.delete(service_group_uuid, no_response=True)
```

Comment:
Again if no response is true then there would be no response. Hence you can create a custom json with msg: "<success message>" and set it in result["response"]